api/app/controllers/spree/api/address_books_controller.rb

```diff
@@ -2,32 +2,41 @@ module Spree
   module Api
     class AddressBooksController < Spree::Api::BaseController
       # Note: the AddressBook is the resource to think about here, not individual addresses
+      before_filter :authorize_user_address
+      before_filter :load_user_addresses
 
       def show
-        render_address_book
+        render :show, status: :ok
       end
 
       def update
         address_params = address_book_params
         default_flag = address_params.delete(:default)
-        address = current_api_user.save_in_address_book(address_params, default_flag)
+        address = address_book_user.save_in_address_book(address_params, default_flag)
         if address.valid?
-          render_address_book
+          render :show, status: :ok
         else
           invalid_resource!(address)
         end
       end
 
       def destroy
-        current_api_user.remove_from_address_book(params[:address_id])
-        render_address_book
+        address_book_user.remove_from_address_book(params[:address_id])
+        render :show, status: :ok
       end
 
       private
 
-      def render_address_book
-        @user_addresses = current_api_user.user_addresses
-        render :show, status: :ok
+      def authorize_user_address
+        authorize! :manage, UserAddress.new(user_id: address_book_user.id)
```

Comment:
I'm fine with the second branch that was mentioned. I'm not _super_ knowledgeable about how all the address stuff works now, but you can also authorize on the `Address` model itself. You can also define a symbol, or use the class (which will get triggered when an instance is used) to represent an address book.

```ruby
can :update, :address_book do |user_id|
  #logic
end
```

But here, I would expect to authorize on the address no? In the destroy action, it's clear we can `can? :destroy, Spree::Address.find(params[:address_id])`

Updating is a very different case here because an update means creation (If I remember correctly) due to the immutable requirement around addresses. In that case I would want to authorize: `can? :create, Spree::Address, user_id: id`

Due to the abstraction it's better I think to keep the two permissions on `Spree::User` (I would just do display and manage), or use a symbol approach like above. The best by far would be to use `Spree::Address`, which is the model actually being created. 

Even further, we could state that if you can update the user, you can update their addresses which lacks granularity and flexibility but makes usage a lot easier. There would only be a check to see if the current user can update the user in question.